app/models/auth_token.rb

```diff
@@ -13,10 +13,26 @@ def data
 
 private
 
-  attr_reader :jwt_token
+  attr_reader :token
 
   def read_token
-    payload, = JWT.decode(jwt_token, secret, true, algorithm: "HS256")
+    read_message_encryptor_token || read_jwt_token
+  end
+
+  def read_message_encryptor_token
+    len = ActiveSupport::MessageEncryptor.key_len
+    key = ActiveSupport::KeyGenerator.new(secret).generate_key("", len)
+    crypt = ActiveSupport::MessageEncryptor.new(key)
+    decrypted_data = crypt.decrypt_and_verify(token)
+    return unless decrypted_data # expired
+
+    JSON.parse(decrypted_data).symbolize_keys
```

Comment:
It seems maybe unnecessary to have this as JSON as I presume the Rails MessageEncryptor is already doing a form of deserialisation?

Reply:
Yes, that's simpler - updated 👍